backend/test/api/mmr_calculation_test.go

```diff
@@ -0,0 +1,236 @@
+		},
+	}
+
+	rr := postRequest("/v2/mmr/calculate", requestBody)
+
+	// Check the status code
+	assert.Equal(t, http.StatusOK, rr.Code)
+
+	// Check the response body
+	var response view.MMRCalculationResponse
+	err := json.Unmarshal(rr.Body.Bytes(), &response)
+	if err != nil {
+		t.Fatalf("Failed to unmarshal response body: %v", err)
+	}
+
+	// Perform assertions on the response
+	assert.Equal(t, 100, response.Team1.Score)
+	assert.Equal(t, 200, response.Team2.Score)
+	assert.Equal(t, 2, len(response.Team1.Players))
+	assert.Equal(t, 2, len(response.Team2.Players))
+
+	// Check Mu and Sigma values for Team 1
+	assert.Equal(t, 28.339272992632903, response.Team1.Players[0].Mu)
+	assert.Equal(t, 6.893615138593818, response.Team1.Players[0].Sigma)
+
+	assert.Equal(t, 26.568046610994696, response.Team1.Players[1].Mu)
+	assert.Equal(t, 6.414912671305059, response.Team1.Players[1].Sigma)
+
+	// Check Mu and Sigma values for Team 2
+	assert.Equal(t, 29.16911282594886, response.Team2.Players[0].Mu)
+	assert.Equal(t, 7.816994122798585, response.Team2.Players[0].Sigma)
+
+	assert.Equal(t, 30.906446819681616, response.Team2.Players[1].Mu)
+	assert.Equal(t, 7.349420941427839, response.Team2.Players[1].Sigma)
+}
+
+// TestSerializationPrecision tests that serialization and deserialization preserves Mu and Sigma values.
+func TestSerializationPrecision(t *testing.T) {
+	// Prepare a sample MMRCalculationRequest with Mu and Sigma as nil
+	originalRequest := view.MMRCalculationRequest{
+		Team1: view.MMRCalculationTeam{
+			Score: 100,
+			Players: []view.MMRCalculationPlayerRating{
+				{Id: 1, Mu: nil, Sigma: nil},
+				{Id: 2, Mu: nil, Sigma: nil},
+			},
+		},
+		Team2: view.MMRCalculationTeam{
+			Score: 200,
+			Players: []view.MMRCalculationPlayerRating{
+				{Id: 3, Mu: nil, Sigma: nil},
+				{Id: 4, Mu: nil, Sigma: nil},
+			},
+		},
+	}
+
+	// Serialize the original request to JSON
+	data, err := json.Marshal(originalRequest)
+	assert.NoError(t, err)
+
+	// Deserialize the JSON back into a new request object
+	var newRequest view.MMRCalculationRequest
+	err = json.Unmarshal(data, &newRequest)
+	assert.NoError(t, err)
+
+	// Check if the original request and new request are equal
+	assert.Equal(t, originalRequest.Team1.Score, newRequest.Team1.Score)
+	assert.Equal(t, originalRequest.Team2.Score, newRequest.Team2.Score)
+
+	for i := range originalRequest.Team1.Players {
+		assert.Equal(t, originalRequest.Team1.Players[i].Id, newRequest.Team1.Players[i].Id)
+		assert.Nil(t, newRequest.Team1.Players[i].Mu)
+		assert.Nil(t, newRequest.Team1.Players[i].Sigma)
+	}
+
+	for i := range originalRequest.Team2.Players {
+		assert.Equal(t, originalRequest.Team2.Players[i].Id, newRequest.Team2.Players[i].Id)
+		assert.Nil(t, newRequest.Team2.Players[i].Mu)
+		assert.Nil(t, newRequest.Team2.Players[i].Sigma)
+	}
+
+	// Now prepare a sample with real Mu and Sigma values
+	realMu := 25.0
+	realSigma := 7.5
+
+	originalRequestWithValues := view.MMRCalculationRequest{
```

Comment:
25.0 and 7.5 would probably not cause precision issues. Are we able to run a calculation and then do precision testing on the resulting Mu and Sigma?